.circleci/run_on_circle.js

```diff
@@ -0,0 +1,167 @@
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'Authorization': 'Basic ' + Buffer.from(program.token).toString('base64'),
+      'Accept': 'application/json',
+      'Host': 'circleci.com',
+    }
+  };
+
+  const body = {
+    branch: program.branch,
+    parameters: {
+      pr_workflow: program.workflow === 'pr',
+      custom_workflow: program.workflow === 'custom',
+      environment: program.environment,
+    }
+  };
+
+  if (program.workflow === 'custom') {
+    body.parameters.browser = program.browser;
+    body.parameters.bver = program.bver;
+    body.parameters.topology = program.topology;
+  }
+
+  return { options, body };
+}
+
+// sends a request using given options/body
+function triggerBuild({options, body}) {
+  console.log('options:', options);
+  console.log('body:', body);
+  return new Promise((resolve, reject) => {
+    const request = http.request(options, function(res) {
+      const chunks = [];
+      res.on('data', chunk => chunks.push(chunk));
+      res.on('end', () => {
+        var resBody = Buffer.concat(chunks);
+        resolve(resBody.toString());
+      });
+    });
+
+    request.once('error', reject);
+    request.end(JSON.stringify(body));
+  });
+}
+
+
+getBranches().then(branches => {
+  const branchPrompt = {
+    type: 'list',
+    name: 'branch',
+    message: 'Branch?',
+    choices: [branches.current, new inquirer.Separator(), ...branches.all],
+    default: branches.current
+  };
+
+  const environmentPrompt = {
+    type: 'list',
+    name: 'environment',
+    message: 'Environment?',
+    choices: ['prod', 'stage', 'dev'],
+    default: 'prod'
+  };
+
+  const workflowPrompt = {
+    type: 'list',
+    name: 'workflow',
+    message: 'Workflow?',
+    choices: ['pr', 'custom'],
+    default: 'pr'
+  };
+
+  const browserPrompt = {
+    when: (answers) => answers.workflow === 'custom',
+    type: 'list',
+    name: 'browser',
+    message: 'Browser?',
+    choices: ['chrome', 'firefox'],
+    default: 'chrome'
+  };
+
+  const confirmPrompt = {
+    type: 'confirm',
+    name: 'confirm',
+    message: 'Confirm the build request?',
+    default: true,
+  };
+
+  const bverPrompt = {
+    when: (answers) => answers.workflow === 'custom',
+    type: 'list',
+    name: 'bver',
+    message: 'Bver ?',
+    choices: ['stable', 'beta', 'unstable'],
+    default: 'stable'
+  };
+
+  const topologyPrompt = {
+    when: (answers) => answers.workflow === 'custom',
+    type: 'list',
+    name: 'topology',
+    message: 'Topology ?',
+    choices: ['group', 'peer-to-peer'],
+    default: 'group'
+  };
+
+  const tokenPrompt = {
+    type: 'input',
+    name: 'token',
+    message: 'Circle CI Token ?',
+    validate: (val) => { return typeof val === 'string' && val.length > 5; }
+  };
+
+  if (process.env.CIRCLE_CI_TOKEN) {
+    tokenPrompt.default = process.env.CIRCLE_CI_TOKEN;
+  }
+
+  inquirer.prompt([
```

Comment:
@makarandp0 I think we should be able to override all these options with a command line flag or an env variable. Then, if all options are provided, we can just show the confirm prompt at the end.

Reply:
@manjeshbhargav - such command line processing would be useful when we invoke something from release tool. But for this tool it's useful to prompt everything - as this is essentially a replacement for the lack  of UI support on circle. I wanted to keep this code very simple.

Reply:
but I agree, once we decide to use this tool from jenkins or other jobs, we would need command line options, and will add those as needed.